bedrock/base/i18n.py

```diff
@@ -0,0 +1,165 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import django.urls
+from django.conf import settings
+from django.utils.text import capfirst
+from django.utils.translation.trans_real import parse_accept_lang_header
+
+from lib.l10n_utils import translation
+
+
+class LocalePrefixPattern(django.urls.LocalePrefixPattern):
+    """
+    A sub-class of Django's "LocalePrefixPattern" that simply normalizes the language
+    prefix for Bedrock (e.g., upper-case country codes).
+
+    This is an essential piece, since it controls the resolution of the
+    language-code prefix of incoming URLs as well as the language code prefix
+    of the URLs generated by Django's "reverse".
+    """
+
+    @property
+    def language_prefix(self):
+        language_code = normalize_language(translation.get_language()) or settings.LANGUAGE_CODE
+        if language_code == settings.LANGUAGE_CODE and not self.prefix_default_language:
+            return ""
+        return f"{language_code}/"
+
+
+def path_needs_lang_code(path):
+    """Convenient way to spot if a path needs a prefix or not"""
+    if path in settings.SUPPORTED_LOCALE_IGNORE:
+        return False
+
+    if path.lstrip("/").split("/")[0] in settings.SUPPORTED_NONLOCALES:
+        return False
+
+    return True
```

Comment:
**note:** It would be nice to make this dynamically check the mozorg nonlocale_urls.py (and to move any other similar routes into a relevant nonlocale_urls.py file.

Note: this comment will make more sense after you've seen the changes to `urls/mozorg_mode.py`